### PR TITLE
Flex Containers, flex-items-per-row

### PR DESCRIPTION
**Pull request by kgcreative**
# Ready
maybe. Please review and make suggestions.

For the default flex container, a 3 way merge may be too heavy, we may do ok with only a 2-way merge

### CODE REVIEW

File: core/utilities/mixins/_flex-container.scss [hunk 1]
```diff
@@ -1,7 +1,7 @@
 @charset "UTF-8";
 
 // Default set of parameters.
-$flex-container-defaults: (
+$default-flex-container: (
```
Comment (kgcreative) [inline core/utilities/mixins/_flex-container.scss]: This follows our naming conventions for other default behaviors

File: core/utilities/mixins/_flex-container.scss [hunk 2]
```diff
@@ -18,7 +18,8 @@ $flex-container-defaults: (
 /// Creates a flexbox container with children
 /// See: https://css-tricks.com/snippets/css/a-guide-to-flexbox for flexbox property explanations.
 ///
-/// selector: *,
+/// parent: '&',
```
Comment (kgcreative) [inline core/utilities/mixins/_flex-container.scss]: Adding parent and children here

File: core/utilities/mixins/_flex-container.scss [hunk 3]
```diff
@@ -38,36 +39,57 @@ $flex-container-defaults: (
...
-  $flex-basis: map-get($settings-map, 'flex-basis');
 
-  #{$parent} {
+  // Then we merge the provided map with the default aggregate from above. This means
+  // that you can specify a container that only has the values that you want to differ
+  // from the default, and everything else is inherited.
+  $_flex-container: map-merge($_default-flex-container, $settings-map);
+
+  // Internal variable parsing
+  $_parent: map-get($_flex-container, parent);
```
Comment (kgcreative) [inline core/utilities/mixins/_flex-container.scss]: we don't need to wrap map-keys in single quotes

Comment (kgcreative) [inline core/utilities/mixins/_flex-container.scss]: Updating _syntax to denote these are internal values

File: core/utilities/mixins/_flex-items-per-row.scss [hunk 4]
```diff
@@ -5,24 +5,25 @@
...
+  $children: '> *',
   $center-grid: true,
   $equal-heights: true,
   $grid: $neat-grid
 ) {
   $_gutter: _retrieve-neat-setting($grid, gutter);
 
   @if ($center-grid == true) and ($equal-heights == true) {
-    @include flex-container($justify-content: center, $align-items: stretch, $selector: $selector);
+    @include flex-container((justify-content: center, align-items: stretch, parent: $parent, children: $children));
```
Comment (kgcreative) [inline core/utilities/mixins/_flex-items-per-row.scss]: Here, we are creating a map on the fly with only the values that we want to override from the default container. I debated between this and creating a map for 'default-centered-equal-height-container' - but that felt too heavy and specific.

File: core/utilities/mixins/_flex-container.scss (hunk 1 above)
Comment (sherakama) [inline core/utilities/mixins/_flex-container.scss]: 🙇 Thanks for this

File: core/utilities/mixins/_flex-container.scss (hunk 2 above)
Comment (sherakama) [inline core/utilities/mixins/_flex-container.scss]: 👍

File: core/utilities/mixins/_flex-container.scss [hunk 5]
```diff
@@ -38,36 +39,41 @@ $flex-container-defaults: (
...
-  $justify-content: map-get($settings-map, 'justify-content');
-  $align-items: map-get($settings-map, 'align-items');
-  $align-content: map-get($settings-map, 'align-content');
-  $flex-grow: map-get($settings-map, 'flex-grow');
-  $flex-shrink: map-get($settings-map, 'flex-shrink');
-  $flex-basis: map-get($settings-map, 'flex-basis');
+  // Merge the provided map with the default aggregate from above. This means
+  // that you can specify a container that only has the values that you want to differ
+  // from the default, and everything else is inherited.
+  $_flex-container: map-merge($default-flex-container, $settings-map);
```
Comment (sherakama) [inline core/utilities/mixins/_flex-container.scss]: I'm more of a fan of the simple merge. If someone overrides the default and messes it up that is their problem and scss should spit out errors in order to get their attention to fix it.

Comment (kgcreative) [inline core/utilities/mixins/_flex-container.scss]: That works for me.

Comment (sherakama) [inline core/utilities/mixins/_flex-container.scss]: woot

File: core/utilities/mixins/_flex-items-per-row.scss (hunk 4 above)
Comment (sherakama) [inline core/utilities/mixins/_flex-items-per-row.scss]: fancy pants

**Comment by sherakama** [comment]
I just did a `npm update` and sass compile after merging this change in to my project and git produced no new changes! That was a smooth update and I hope we can continue with this.

**Comment by kgcreative** [comment]
Woot!

**Pull request merged**